docs/developer/architecture.rst

```diff
@@ -64,7 +64,9 @@ For the simplicity the other backend systems - Dask and MPI are omitted and only
 
 Component View
 --------------
-Coming soon...
+.. toctree::
+   flow/engines/base/frame/index
+
 
 Module/Class View
```

Comment:
we should insert file tree here as well, or wait for https://github.com/modin-project/modin/pull/2952 to land first

Reply:
Let's merge #2952 first